frontend/src/components/input-output/list-of-sources/ListOfSources.jsx

```diff
@@ -109,6 +109,7 @@ function ListOfSources({ setSelectedSourceId, type }) {
           placeholder="Search"
           prefix={<SearchOutlined className="search-outlined" />}
           onChange={onSearchDebounce}
+          value={searchText}
```

Comment:
@athul-rs I'm a bit unclear on how adding the `value` prop fixed the search issue. Please correct me if I'm wrong. The search results (filtered list of adapters) will still remain the same across all adapters. The only change that we're doing here is persisting the search text, even after the user closes and reopens the modal. So essentially, the filtering of results remains the same as before, but now the search text will be displayed in the search field for better clarity.

Reply:
yeah, but it provides more context to why the same is displayed.
For example;
- On searching "azure", even though the search remains the same across multiple modals in the same setup (because of current setup using `useEffect` ), by adding the `value` prop it will show the search bar with the value which wasn't happening before.

Providing more context to user.

Should we clear out the field each time the modal closes ? instead ? Suggestions here please : @tahierhussain 
My thought was this would be a better solution, instead of typing something similar everytime. cc : @nehabagdia

Reply:
@athul-rs While this is a great point you have made, I am leaning towards clearing the search text on each modal close. 
Let us revisit this based on customer traction at a later stage.